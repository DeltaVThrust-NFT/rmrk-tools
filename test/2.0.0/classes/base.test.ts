import { cryptoWaitReady } from "@polkadot/util-crypto";
import { getBobKey, createBaseMock, createNftClassMock } from "../mocks";

beforeAll(async () => {
  return await cryptoWaitReady();
});

describe("rmrk2.0.0 Base: Base", () => {
  it("should match snapshot", async () => {
    const baseClass = createBaseMock();
    expect(await baseClass.base()).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 Base: Base", () => {
  it("should throw error", async () => {
    const baseClass = createBaseMock(1);
    try {
      await baseClass.base();
    } catch (e) {
      expect(e.message).toMatch("An already existing BASE cannot be minted!");
    }
  });
});

describe("rmrk2.0.0 Base: Change issuer", () => {
  it("should match snapshot", async () => {
    const baseClass = createBaseMock(1);
    expect(
      await baseClass.change_issuer(getBobKey().address)
    ).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 Base: Base", () => {
  it("should throw error", async () => {
    const baseClass = createBaseMock(0);
    try {
      await baseClass.change_issuer(getBobKey().address);
    } catch (e) {
      expect(e.message).toMatch(
        "This base is new, so there's no issuer to change. If it has been deployed on chain, load the existing nft class as a new instance first, then change issuer."
      );
    }
  });
});
