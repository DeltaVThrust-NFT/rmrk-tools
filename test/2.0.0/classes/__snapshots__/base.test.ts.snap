// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rmrk2.0.0 Base: Base should match snapshot 1`] = `"RMRK::BASE::2.0.0::%7B%22symbol%22%3A%22KBASE777%22%2C%22type%22%3A%22svg%22%2C%22issuer%22%3A%225GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY%22%2C%22parts%22%3A%5B%7B%22id%22%3A%22background%22%2C%22type%22%3A%22slot%22%2C%22equippable%22%3A%5B%5D%2C%22z%22%3A0%7D%2C%7B%22id%22%3A%22backpack%22%2C%22type%22%3A%22slot%22%2C%22equippable%22%3A%5B%22d43593c715a56da27d-KANARIABIRDS%22%5D%2C%22z%22%3A1%7D%2C%7B%22id%22%3A%22tail%22%2C%22type%22%3A%22fixed%22%2C%22src%22%3A%22ipfs%3A%2F%2Fipfs%2FQmcEuigDVCScMLs2dcrJ8qU4Q265xGisUyKeYdnnGFn6AE%2Fvar3_tail.svg%22%2C%22z%22%3A2%7D%5D%7D"`;

exports[`rmrk2.0.0 Base: Change issuer should match snapshot 1`] = `"RMRK::CHANGEISSUER::2.0.0::base-1-KBASE777::5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"`;
